src/libcore/str/lossy.rs

```diff
@@ -70,6 +70,7 @@ impl<'a> Iterator for Utf8LossyChunksIter<'a> {
                 let w = core_str::utf8_char_width(byte);
 
                 macro_rules! error { () => ({
+                    // SAFETY: we have checked up to `i` that source is valid UTF-8
```

Comment:
Note: I didn't actually verify that this is true. The structure makes sense, though.